units/src/parse.rs

```diff
@@ -225,8 +225,8 @@ pub fn hex_u32_unprefixed(s: &str) -> Result<u32, UnprefixedHexError> {
 ///
 /// # Errors
 ///
-/// - If input string is not a valid hex encoding of a `u32`.
-/// - With `InvalidDigit` due to the `x` if there is a prefix.
+/// - If the input string contains a `0x` (or `0X`) prefix, with `InvalidDigit` due to the `x`.
+/// - If the input string is not a valid hex encoding of a `u32`.
 pub fn hex_u32_unchecked(s: &str) -> Result<u32, ParseIntError> {
```

Comment:
Hmm, why is it call unchecked? I don't see anything that needs checking and the doc doesn't explain it.

Reply:
It is used by the `hex_u32_unprefixed()` and `hex_u32_prefixed()` after first checking for a prefix.  i.e. compared to these two functions it is "unchecked".  I added to the doc that it doesn't first check for a prefix to try to clarify this.

Reply:
Looks like they should've been private...

Reply:
That's what I thought when I looked at  it just then.  I also just changed it for u128 too.

Reply:
We use `hex_u128_unchecked` in `bitcoin::pow` so for API uniformity it seems correct to have this function public.

Reply:
@tcharding that sounds really weird, I'll need to investigate.